lib/testy.js

```diff
@@ -176,15 +181,39 @@ class Testy {
     }
   }
 
-  async #loadFileHandlingErrors(file) {
+  async #loadFileHandlingErrors(filePath) {
+    let importPath = filePath;
+    let isTempFileCreated = false;
+
     try {
-      testRunner.loadingFile(file);
-      await import(file);
+      testRunner.loadingFile(filePath);
+      const fileExtension = path.extname(filePath);
+
+      if (fileExtension === '.ts') {
+        const transpilationResult = await TypescriptTranspiler.generateTemporalJavascriptFromTypescript(filePath);
+        importPath = transpilationResult.importPath
+        isTempFileCreated = transpilationResult.isTempFileCreated
+      }
+
+      const fileUrl = pathToFileURL(importPath);
+      await import(fileUrl);
     } catch (err) {
       ui.exitWithError(
-        I18nMessage.of('error_loading_suite', file), errorDetailOf(err),
-        I18nMessage.of('feedback_for_error_loading_suite'),
+          I18nMessage.of('error_loading_suite', filePath), errorDetailOf(err),
+          I18nMessage.of('feedback_for_error_loading_suite'),
       );
+    } finally {
+      this.#cleanupTemporaryFile(isTempFileCreated, importPath)
+    }
+  }
+
+  #cleanupTemporaryFile(shouldRemoveFile, path) {
+    if (shouldRemoveFile && path && fs.existsSync(path)) {
+      try {
+        fs.unlinkSync(path);
+      } catch (cleanupError) {
+        console.error(`[Testy] Warning: Failed to delete temporary file ${path}:`, cleanupError);
```

Comment:
We are not accessing console directly, we have ConsoleUI as an intermediate interface. Could we move this warning report there?